src/secure_wipe.rs

```diff
@@ -0,0 +1,104 @@
+use std::{
+    fmt,
+    ops::Deref,
+    ptr::write_volatile,
+    sync::atomic::{compiler_fence, Ordering::SeqCst},
+};
+
+use rand::{thread_rng, Rng};
+use serde::{Deserialize, Serialize};
+
+/// A `Vec<u8>` that gets scrambled on drop.
+/// Provides a vector that will scramble allocations as
+/// they are discarded when the vector grows.
+///
+/// Allows users to scamble the data whenever they want.
+///
+/// Also scrambles data on Drop.
+///
+/// Uses the basic idea (`write_volatile` + `compiler_fence`)
+/// from @bascule's zeroize crate but overwrites data with
+/// random bytes instead of zeros.
+#[derive(Clone, Default, Deserialize, Serialize)]
+pub(crate) struct SecureVec(Vec<u8>);
+
+impl SecureVec {
+    pub(crate) fn scramble(&mut self) {
+        let mut rng = thread_rng();
+        for byte in &mut self.0 {
+            #[allow(unsafe_code)]
+            unsafe {
+                write_volatile(byte, rng.gen());
+            }
+        }
+        compiler_fence(SeqCst);
+    }
+}
```

Comment:
this is broken without the push function that I included originally

Reply:
when parser::expect_info reads in the ServerInfo, it may realloc, causing non-scrambled copies of memory to float around in the vacated allocations